File: Problems/Algorithms/462. Minimum Moves to Equal Array Elements II/min_moves_equal_elements_II.go
import (
    "sort"
)

func minMoves2(nums []int) int {
    n := len(nums)

    sort.Ints(nums)

    var median int
    median = nums[n/2]

    ans := 0
    for _, num := range nums {
        if num >= median {
            ans += num - median
        } else {
            ans += median - num
        }
    }
    return ans
}

File: Problems/Algorithms/462. Minimum Moves to Equal Array Elements II/min_moves_equal_elements_II.rs
impl Solution {
    pub fn min_moves2(mut nums: Vec<i32>) -> i32 {
        nums.sort();
        let n = nums.len();
        nums.iter().fold(0, |mut acc, x| {
            acc += if x >= &nums[n/2] { x - &nums[n/2] } else { &nums[n/2] - x };
            acc
        })
    }
}

